spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/condition/OnMissingServletFilterCondition.java

```diff
@@ -0,0 +1,162 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.Filter;
+
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.ConfigurationCondition;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+import org.springframework.core.type.MethodMetadata;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.ReflectionUtils;
+
+/**
+ * {@link Condition} that checks for the absence of specific Servlet Filters.
+ * @author Brian Clozel
+ * @since 1.5.0
+ */
+@Order(Ordered.LOWEST_PRECEDENCE)
+public class OnMissingServletFilterCondition extends SpringBootCondition implements ConfigurationCondition {
+
+	@Override
+	public ConfigurationPhase getConfigurationPhase() {
+		return ConfigurationPhase.REGISTER_BEAN;
+	}
+
+	@Override
+	public ConditionOutcome getMatchOutcome(ConditionContext context, AnnotatedTypeMetadata metadata) {
+		ConditionMessage matchMessage = ConditionMessage.empty();
+		if (metadata.isAnnotated(ConditionalOnMissingServletFilter.class.getName())) {
+			List<String> servletFilters = parseAnnotationForServletFilters(context, metadata);
+			Set<String> matching = getMatchingBeans(context, servletFilters);
+			if (matching.isEmpty()) {
+				return ConditionOutcome.match(ConditionMessage
+						.forCondition(ConditionalOnMissingServletFilter.class)
+						.didNotFind("any Servlet filter").atAll());
+			}
+			else {
+				return ConditionOutcome.noMatch(ConditionMessage
+						.forCondition(ConditionalOnMissingServletFilter.class)
+						.found("servlet filter", "servlet filters")
+						.items(ConditionMessage.Style.QUOTE, matching));
+			}
+		}
+		return ConditionOutcome.match(matchMessage);
+	}
+
+	private List<String> parseAnnotationForServletFilters(ConditionContext context, AnnotatedTypeMetadata metadata) {
+		List<String> servletFilters = new ArrayList<String>();
+		String annotationType = ConditionalOnMissingServletFilter.class.getName();
+		List<Object> values = metadata
+				.getAllAnnotationAttributes(annotationType, true).get("value");
+		if (values != null) {
+			for (Object value : values) {
+				if (value instanceof String[]) {
+					Collections.addAll(servletFilters, (String[]) value);
+				}
+				else {
+					servletFilters.add((String) value);
+				}
+			}
+		}
+		if (servletFilters.isEmpty() &&
+				metadata instanceof MethodMetadata && metadata.isAnnotated(Bean.class.getName())) {
+			addDeducedBeanTypeForBeanMethod(context, metadata, servletFilters, (MethodMetadata) metadata);
```

Comment:
There is only a single use where the value is omitted. Perhaps it's not a frequent use case? If it isn't, maybe it would be nicer to just not support it and remove all the code that deduces the type?